projects/plugins/boost/.eslintrc.cjs

```diff
@@ -3,45 +3,30 @@ const loadIgnorePatterns = require( 'jetpack-js-tools/load-eslint-ignore.js' );
 
 module.exports = {
 	root: true,
-	parser: '@typescript-eslint/parser',
 	extends: [
-		// '@sveltejs',
 		require.resolve( 'jetpack-js-tools/eslintrc/base' ),
-		'plugin:@typescript-eslint/recommended',
-		'plugin:@wordpress/eslint-plugin/recommended',
+		require.resolve( 'jetpack-js-tools/eslintrc/wp-eslint-plugin/recommended' ),
+		require.resolve( 'jetpack-js-tools/eslintrc/typescript' ),
```

Comment:
I think we should extend `require.resolve( 'jetpack-js-tools/eslintrc/typescript' )` in these two files as well:
- `projects/plugins/jetpack/.eslintrc.js`
- `projects/js-packages/components/.eslintrc.cjs`

Or may be extend it in base config itself to make it available globally because it will be used only for TS files.

Reply:
We can override those rules or use overrides to apply the rules only to `.ts` and `.tsx` files.

Simply put, we need that TS related ESLint config wherever we use TS, otherwise we will end up having many unwanted lints like this

<img width="1307" alt="Screenshot 2022-05-25 at 9 57 43 AM" src="https://user-images.githubusercontent.com/18226415/170180010-71d42116-e7a8-499d-9c18-daec7c15bc8d.png">

<img width="1138" alt="Screenshot 2022-05-25 at 10 01 09 AM" src="https://user-images.githubusercontent.com/18226415/170180145-76ac17a0-4e3a-4a50-bf78-bf88cdc8006c.png">

Reply:
If we can merge it into the base with overrides so the TS stuff doesn't interfere with the non-TS stuff, I'd be happy to go with that.

Reply:
Add this simple override in `js-tools/eslintrc/base.js` should do the job:

```js
overrides: [
	{
		files: [ '*.ts', '*.tsx' ],
		extends: './typescript',
	},
],
```

After adding that, we don't need to extend `typescript` config here in `plugins/boost/.eslintrc.cjs`
We may also be able to remove `@typescript-eslint/no-var-requires` override for root `.cjs` files.

Reply:
If I do this in `js-tools/eslintrc/typescript.js`

```diff
diff --git a/tools/js-tools/eslintrc/typescript.js b/tools/js-tools/eslintrc/typescript.js
index 4df6d88f9a..5b9ca0738c 100644
--- a/tools/js-tools/eslintrc/typescript.js
+++ b/tools/js-tools/eslintrc/typescript.js
@@ -12,12 +12,15 @@ module.exports = {
                'no-undef': 0,
 
                '@typescript-eslint/no-unused-vars': [ 'warn', { argsIgnorePattern: '^_' } ],
+               '@typescript-eslint/ban-ts-comment': 0,
        },
        overrides: [
                {
-                       files: [ '*.cjs' ],
+                       files: [ '*.js', '*.cjs', '*.jsx' ],
                        rules: {
                                '@typescript-eslint/no-var-requires': 0,
+                               '@typescript-eslint/no-empty-function': 0,
+                               '@typescript-eslint/no-this-alias': 0,
                        },
                },
        ],
```

And then directly extend `./typescript` in `base.js`, then there are no lints left.

Is that something we can do?

Reply:
I think I like your suggestion in https://github.com/Automattic/jetpack/pull/24355#discussion_r881781346 better than the one in https://github.com/Automattic/jetpack/pull/24355#discussion_r883502830

The first suggestion limits typescript checking to just the typescript files, which seems good. It would also let us remove the override in projects/plugins/boost/.eslintrc.cjs since those would no longer be being checked as typescript either.

When I try it locally, it reports four new errors:

```
/usr/local/src/automattic/jetpack/projects/js-packages/components/components/text/types.ts
  45:44  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  46:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/usr/local/src/automattic/jetpack/projects/plugins/boost/app/assets/src/js/elements/Toggle.svelte
  2:14  error  Parsing error: Missing semicolon. (3:13)

/usr/local/src/automattic/jetpack/projects/plugins/boost/app/assets/src/js/pages/settings/elements/RatingCard.svelte
  28:11  error  'Jetpack_Boost' is not defined  no-undef
```

The first two seem to be because js-packages/components isn't doing the typescript lints yet. The third seems to be because the svelte file has embedded typescript but the typescript parser isn't being used. The last is because a rule supposedly disabled just for typescript was accidentally applying to the svelte files too.

That last is why I don't care for your second comment's suggestion, as it would apply those to all files rather than only to typescript files.

P.S. In either case, we could probably just merge the config into base.js instead of having a separate typescript.js.

Reply:
Alright, thank you for your thoughts. Here is #24579 to fix the issue.